src/FSharpPlus/Extensions/Task.fs

```diff
@@ -91,6 +91,49 @@ module Task =
                         ) |> ignore) |> ignore
             tcs.Task
 
+    /// <summary>Creates a task workflow from three workflows 'x', 'y' and z, mapping its results with 'f'.</summary>
+    /// <remarks>Workflows are run in sequence.</remarks>
+    /// <param name="f">The mapping function.</param>
+    /// <param name="x">First task workflow.</param>
+    /// <param name="y">Second task workflow.</param>
+    /// <param name="z">Third task workflow.</param>
+    let map3 (f : 'T -> 'U -> 'V -> 'W) (x : Task<'T>) (y : Task<'U>) (z: Task<'V>) : Task<'W> =
```

Comment:
Could this be rewritten to something like below?
```fsharp
    let map3 (f : 'T -> 'U -> 'V -> 'W) (x : Task<'T>) (y : Task<'U>) (z: Task<'V>) : Task<'W> =
        x |> bind (fun x' -> y |> bind (fun y' -> z |> map (fun z' -> f x' y' z')))
```

also `map2` would be

```fsharp
    let map2 (f : 'T -> 'U -> 'V) (x : Task<'T>) (y : Task<'U>) : Task<'V> =
        x |> bind (fun x' -> y |> map (fun y' -> f x' y'))
```

it would require us to move bind and join above `map2`/`map3`

Reply:
similar approach could be applied to other maps (to places where it would simplify code)

Reply:
The resulting code would be equivalent but I'm not sure it will be as efficient, since it will allocate 3 TaskCompletionSources insted of 1, and that alone would defeat the whole purpose of having a specific implementation for `map3`.

So, unless you can prove there's no performance penalty I would leave it as it is.

Reply:
Yes, it will allocate more. I agree that we should not change it.